R/middleware.R

```diff
@@ -311,16 +311,32 @@ HandlerManager <- R6Class("HandlerManager",
         },
         call = .httpServer(
           function (req) {
-            withCallingHandlers(withLogErrors(handlers$invoke(req)),
-              error = function(cond) {
-                sanitizeErrors <- getOption('shiny.sanitize.errors', FALSE)
-                if (inherits(cond, 'shiny.custom.error') || !sanitizeErrors) {
-                  stop(cond$message, call. = FALSE)
-                } else {
-                  stop(paste("An error has occurred. Check your logs or",
-                             "contact the app author for clarification."),
-                       call. = FALSE)
+            hybrid_chain(
+              hybrid_chain(
+                withCallingHandlers(withLogErrors(handlers$invoke(req)),
+                  error = function(cond) {
+                    sanitizeErrors <- getOption('shiny.sanitize.errors', FALSE)
+                    if (inherits(cond, 'shiny.custom.error') || !sanitizeErrors) {
+                      stop(cond$message, call. = FALSE)
+                    } else {
+                      stop(paste("An error has occurred. Check your logs or",
+                                 "contact the app author for clarification."),
+                           call. = FALSE)
+                    }
+                  }
+                ),
+                catch = function(err) {
+                  httpResponse(status = 500L,
+                    content_type = "text/html",
+                    content = as.character(htmltools::htmlTemplate(
+                      system.file("template", "error.html", package = "shiny"),
+                      message = conditionMessage(err)
+                    ))
+                  )
                 }
+              ),
+              function(resp) {
+                maybeInjectAutoreload(resp)
```

Comment:
Will this only inject the autoreload stuff into the index page for an app? I just want to make sure it doesn't get injected into any other pages that happen to get served up.

Reply:
I think it might be for every page with `</head>`. I was thinking it couldn't just be `/` because of Shiny Rmd. I don't think adding the script everywhere is harmful though? (Keep in mind this is just for development use)

Reply:
Actually I guess we could move to this wherever we insert `<script src="shiny.js">` or whatever?

Reply:
Wait, sorry, the whole reason for putting it here is it needs to apply to error message pages too.